lib/storage-service.js

```diff
@@ -111,8 +111,17 @@ class StorageService {
    */
   _getLeaseManager(token, secret, ProviderType) {
     const secretID = `/${ProviderType.prototype.constructor.name}/${token}/${secret}`;
+    let tokenManager;
+
+    if (typeof ProviderType.prototype.requireVaultToken === 'function') {
```

Comment:
My rationale here is that we're passing around the provider type rather than instances of the provider. We need a way to check whether a provider needs a Vault token before we instantiate it. An option would be to use a getter and test for the presence of that property but it's basically the same method.